addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -126,10 +176,12 @@ class ComposerTextInput extends Component {
      */
     _update() {
         this._textareaRef.el.value = this.composer.textInputContent;
-        this._textareaRef.el.setSelectionRange(
-            this.composer.textInputCursorStart,
-            this.composer.textInputCursorEnd
-        );
+        if (!this.state.isMentionsOpen) {
+            this._textareaRef.el.setSelectionRange(
+                this.composer.textInputCursorStart,
+                this.composer.textInputCursorEnd
+            );
+        }
```

Comment:
this looks wrong, pretty sure this is why the cursor position is buggy when mention menu is open